Features/layout console helper

ferndopolis opened the pull request:
WIP, for review only 
creates the timecop_console basic layout with a call to timecop_console_layout helper

ferndopolis committed:
add timecop_console_layout helper

helper method builds the basic layout needed to control your app server
time, method can also take in a block of html which is then inserted at
the top of the timecop console.

michaelkirk commented on app/helpers/timecop_console/main_helper.rb:
```diff
@@ -17,5 +17,23 @@ def time_travel_to(date)
     end
+
+    def timecop_console_layout
+      if Rails.env.development? || Rails.env.staging?
```
I think you should get rid of these checks, and expect that the person _using_ this helper will decide when they want to render the controls. E.g. a lot of projects don't even have a "staging" environment.

E.g. in my application.rb
    <%= timecop_console_layout if Rails.env.development? || Rails.env.staging? %>

michaelkirk commented on app/helpers/timecop_console/main_helper.rb:
```diff
@@ -17,5 +17,23 @@ def time_travel_to(date)
+    def timecop_console_layout
+      if Rails.env.development? || Rails.env.staging?
+        content_tag(:div, id: "debug-console") do
+          concat(yield) if block_given?
```
A lot of view code here. Instead of building all these DOM elements by hand, can you render an external view template?

ferndopolis committed:
move the timecop console layout to view

ferndopolis commented:
@michaelkirk, yeah good point i guess my intent here was for adding some dynamic content rendering in the future, but currently only yielding a block is it!!! those content_tag and concat are nasty.  

did you push up your changes to the dummy app so it we can access the index page for QAing?

ferndopolis committed:
fix up timecop_console_layout after rebase

ferndopolis committed:
refactor hour, min assignments

ferndopolis committed:
make call to console_layout partial more explicit

ferndopolis committed:
add more specs to time_travel_to

ferndopolis committed:
clean up configs

sprokets/railtie add to include assets:precompile rake tasks
add debug support for precompiled assets
update Bundler config per rails 4 upgrade guide

ferndopolis commented:
@michaelkirk ok this one is ready for another review. there is a bunch of changes some that could have been in another PR
- move the layout content to a partial view.
- add name and DateTime object passing to time_travel_to
- more spec for time_travel_to
- updated the config file for dummy app

somethings more things I am going to do:
- [ ] come up with some useful time_travel_to buttons to add to the layout (ie 1.hour.from_now)
- [ ] style the layout 
- [ ] add ability to pass a block to timecop_console_layout so users can add their own button within the timecop console div --> will complete this to another branch
- [ ] update the README.rdoc

michaelkirk commented on app/helpers/timecop_console/main_helper.rb:
```diff
@@ -1,21 +1,27 @@
       end
 
+      name ||= date.strftime("%B %d, %Y")
+      hour = date.respond_to?(:hour) ? date.hour : 12
```
:bird: quack

michaelkirk commented:
looks good!

ferndopolis commented:
@michaelkirk, i notice that now our test environment wont launch due to error with sprokets/railties 
ahgeez the pain of upgrading dummy gem to rails4!!!

michaelkirk commented:
Do you mean you can't run "bundle exec rspec" ?

On Thu Mar  6 14:14:09 2014, Fernando Marcheselli wrote:

> @michaelkirk https://github.com/michaelkirk, i notice that now our
> test environment wont launch do to error with sprokets/railties
> ahgeez the pain of upgrading dummy gem to rails4!!!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ferndopolis/timecop-console/pull/6#issuecomment-36944099.

ferndopolis commented:
yes siree

On Thu, Mar 6, 2014 at 2:20 PM, Michael Kirk notifications@github.comwrote:

> Do you mean you can't run "bundle exec rspec" ?
> 
> On Thu Mar 6 14:14:09 2014, Fernando Marcheselli wrote:
> 
> > @michaelkirk https://github.com/michaelkirk, i notice that now our
> > test environment wont launch do to error with sprokets/railties
> > ahgeez the pain of upgrading dummy gem to rails4!!!
> > 
> > ## 
> > 
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/ferndopolis/timecop-console/pull/6#issuecomment-36944099
> > .
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/ferndopolis/timecop-console/pull/6#issuecomment-36944818
> .

ferndopolis committed:
clean up the timecop-console layout

for now returned the layout to be without any time_travel_to buttons b/c
so that the users can add there own buttons and not clutter the console

ferndopolis committed:
require sprockets in dummy app development env

running specs on the dummy app resulted in a failing to launch the test
environment with the sprockets library required a quick solution was to
remove it from the test environment

ferndopolis committed:
update README.rdoc
